In [ ]:
import sys 
print(sys.version) 
print(sys.executable) 

3.12.14 | packaged by Anaconda, Inc. | (main, Aug 27 2026, 14:37:13) [MSC v.1942 64 bit (AMD64)]
C:\Users\Aphilox\anaconda3\envs\ICD341\python.exe


In [ ]:
import numpy as np 
import pandas as pd 
import matplotlib.pyplot as plt 
print("Scientific Python environment working!")

Scientific Python environment working!


# Lab 1 Answers

**Q1: What is a Python environment?**
An isolated workspace containing a specific Python interpreter and specific package versions.

**Q2: Why is it useful to create a separate environment for a course?**
It prevents package conflicts, keeps the course reproducible, and lets you delete it without affecting other projects.

**Q3: What does conda activate icd341 do?**
It switches your terminal to use the icd341 environment’s Python and packages.

**Q4: What is the purpose of specifying python=3.12?**
It installs Python 3.12 inside the environment, ensuring compatibility and reproducibility.

**Q5: What does import numpy as np mean?**
It imports the NumPy module and gives it the alias np.

**Q6: Why are aliases such as np commonly used?**
They are shorter, conventional, and reduce typing.

**Q7: What happens if you try to import a package that isn't installed?**
You get a ModuleNotFoundError.

**Q8: Explain the difference between sys.version and sys.executable.**
sys.version gives the Python version and build information. sys.executable gives the full path to the Python interpreter running the notebook.

**Q9: Define a Python environment.**
An isolated set of Python packages and interpreter versions.

**Q10: State two advantages of using Conda environments.**
Isolation from other projects; easy reproducibility via environment.yml.

**Q11: What is JupyterLab?**
A web-based interactive development environment for notebooks, code, and data.

**Q12: What is VS Code?**
A lightweight but powerful code editor/IDE with extensions.

**Q13: What is Git?**
A distributed version control system for tracking code changes.

**Q14: What is GitHub?**
A cloud platform for hosting Git repositories and collaborating.

**8Q15: Explain the difference between git add and git commit.**
git add stages changes; git commit saves staged changes into history.

**Q16: What is the purpose of .gitignore?**
Tells Git which files/folders to ignore.

**Q17: Why should .ipynb_checkpoints normally be excluded from Git?**
They are auto-generated backup files, not source code, and cause noise/conflicts.

**Q18: Explain why reproducible environments are important in statistical computing.**
They ensure the same package versions and behaviour, so results can be reproduced by you and others.
